### ISSUE TITLE
[FEA] Add an insert function that takes separate key/value iterators.

### ISSUE THREAD

**Issue by seunghwak** [NVIDIA/cuCollections#84]
https://github.com/NVIDIA/cuCollections/blob/8fce363b40254f4170d97eaedcb12fc8e84e1b74/include/cuco/static_map.cuh#L200

currently takes an iterator of zipped (key, value) pairs. This requires additional work at the caller side.

something like
```
void insert(KeyIt key_first, KeyIt key_last, ValueIt value_first, Hash hash = Hash{}, KeyEqual key_equal = KeyEqual{});
```
will be more convenient.

**Comment by jrhemstad** [comment]
Hm, so I went ahead and did this and it ends up causing an ambiguous overload situation because of the defaulted arguments, e.g., https://godbolt.org/z/6ThvPohej

I'll have to ponder some more how to make this work.

**Comment by seunghwak** [comment]
I see, and this is not a high priority task at all. It seems like cuco gets updated to take thrust::tuple (instead of requiring thrust::pair), and now I just need to use zip_iterator instead of combining zip_iterator & transform_iterator.

https://github.com/rapidsai/cugraph/pull/1640/files#diff-d7a948ed123361640cbf30b0b3be8ea6c3dae94c026e538c62d68626cca90383L205-R205

**Comment by jrhemstad** [comment]
Yeah, in https://github.com/NVIDIA/cuCollections/pull/79 I updated the `pair` internals to work with any "pair-like" thing. That includes `thrust::tuple` or a zip iterator.
